Overhaul AVM FRITZ!Box Tools docs (#35589)
home-assistant/home-assistant.github.io: source/_integrations/fritz.markdown

ha_integration_type: integration
---

The AVM FRITZ!Box Tools integration allows you to control your [AVM FRITZ!Box](https://en.avm.de/products/fritzbox/) router and have presence detection for connected network devices.

There is support for the following platform types within Home Assistant:

- **{% term "Device tracker" %}** - presence detection by looking at connected devices.
- **{% term "Binary sensor" %}** - connectivity status.
- **{% term Image %}** - QR code for Guest Wi-Fi.

> - **{% term Button %}** - reboot, reconnect, firmware_update.

Reviewer: ```suggestion
- **{% term Button %}** - reboot, reconnect, firmware update.
```

- **{% term Sensor %}** - external IP address, uptime and network monitors.
- **{% term Switch %}** - call deflection, port forward, parental control and Wi-Fi networks.
- **{% term Update %}** - firmware status of the device.

{% include integrations/config_flow.md %}

{% important %}
Both the TR-064 (_Permit access for apps_) and UPnP (_Transmit status information over UPnP_) protocol needs to be enabled in the FRITZ!Box under **Home Network** > **Network** > **Network settings** > **Access Settings in the Home Network** for Home Assistant to login and read device info.
{% endimportant %}

## Username

> We strongly recommend to create a separate user to connect Home Assistant to your FRITZ!Box. This can be done under int the FRITZ!Box under **System** > **FRITZ!Box Users** > **Users** > **Add User**. The user needs the **FRITZ!Box Settings** permission.

Reviewer: ```suggestion
It is recommended to create a separate user to connect Home Assistant to your FRITZ!Box. To create a user, in the FRITZ!Box go to **System** > **FRITZ!Box Users** > **Users** > **Add User**. Make sure the user has the **FRITZ!Box Settings** permission.
```

{% note %}

> If you still want to use the predefined user, please note that starting from FRITZ!OS 7.24 the FRITZ!Box creates a random username for the admin user if you didn't set one yourself. This can be found after logging into the FRITZ!Box and visiting **System** > **FRITZ!Box Users** > **Users**. The username starts with `fritz` followed by four random numbers. Under properties on the right it says `created automatically`. Prior to FRITZ!OS 7.24 the default username was `admin`.

Reviewer: ```suggestion
If you still want to use the predefined user, please note that as of FRITZ!OS 7.24, the FRITZ!Box creates a random username for the admin user if you didn't set one yourself. This can be found after logging into the FRITZ!Box and visiting **System** > **FRITZ!Box Users** > **Users**. The username starts with `fritz` followed by four random numbers. Under properties on the right it says `created automatically`. Prior to FRITZ!OS 7.24, the default username was `admin`.
```

It is possible to change some behaviors through the integration options.
To change the settings, go to {% my integrations title="**Settings** > **Devices & services**" %}. Select the **AVM FRITZ!Box Tools** integration, then select **Configure**.

- **Consider home**: Number of seconds that must elapse before considering a disconnected device "not at home".
- **Enable old discovery method**: Needed on some scenarios like no mesh support (fw <= 6.x), mixed brands network devices or LAN switches.
### Consider home

Number of seconds that must elapse before considering a disconnected device "not at home".

## Additional info
### Enable old discovery method

### Parental control
Needed on some scenarios like no mesh support (_FritzOS <= 6.x_) or mixed brands network devices or LAN switches.

Parental control switches can be used to enable and disable internet access of individual devices. If a device has internet access it will be enabled, otherwise it will be disabled. You can also find the current blocking state of the individual devices in the UI of the FRITZ!Box under `Internet` -> `Filters` -> `Parental Controls` -> `Device Block`
## Additional information

> Parental control switches are designed for advanced users, thus they are disabled by default. You need to enable the wanted entities manually.

Author: this did not correspond to the implementation, therefore it is removed

          message: "Password: ..."

```

## Troubleshooting

In any case, when reporting an issue, please enable [debug logging](/docs/configuration/troubleshooting/#debug-logs-and-diagnostics), restart the integration, and as soon as the issue re-occurs stop the debug logging again (_download of debug log file will start automatically_). Further _if still possible_, please also download the {% term diagnostics %} data. If you have collected the debug log and the diagnostics data, provide them with the issue report.

### Device presence detection is not working as expected

> You see a device as still present, even it is offline or disconnected for more than the configured [consider home](#consider-home) seconds? You're using additional network equipment like a network switch or wifi access point other than an AVM Fritz!Repeater or other AVM components, but not configured as a [mesh](https://en.avm.de/service/knowledge-base/dok/FRITZ-Box-7590/3329_Mesh-with-FRITZ/) in your home network?

Reviewer: ```suggestion
Check if one of the following cases applies:

- You see a device as still present, even if it is offline or disconnected for more than the configured [consider home](#consider-home) seconds.
- You're using additional network equipment like a network switch or Wi-Fi access point other than an AVM Fritz!Repeater or other AVM components, but not configured as a [mesh](https://en.avm.de/service/knowledge-base/dok/FRITZ-Box-7590/3329_Mesh-with-FRITZ/) in your home network.
```

> Than you should [enable the old discovery method](#enable-old-discovery-method) in the [integration options](#integration-options).

Reviewer: ```suggestion
If one of the above cases applies to your setup, try [enabling the old discovery method](#enable-old-discovery-method) in the [integration options](#integration-options). This might resolve the issue.
```
